Algorithm.Framework/Selection/OpenInterestFutureUniverseSelectionModel.cs

```diff
@@ -0,0 +1,138 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.Farmework.Selection
+{
+    /// <summary>
+    ///     Selects contracts in a futures universe, sorted by open interest.  This allows the selection to identifiy current
+    ///     active contract.
+    /// </summary>
+    public class OpenInterestFutureUniverseSelectionModel : FutureUniverseSelectionModel
```

Comment:
What do you mean by that sorry?

Reply:
Like this guy `BasicTemplateFuturesFrameworkAlgorithm` https://github.com/QuantConnect/Lean/blob/master/Algorithm.CSharp/BasicTemplateFuturesFrameworkAlgorithm.cs, a small simple algorithm which uses open-source available data to assert the behavior of the selection model and is an example of it usage.
*No need to define a custom alpha or PortfolioConstructionModel as `BasicTemplateFuturesFrameworkAlgorithm` does, just using the new `OpenInterestFutureUniverseSelectionModel` and placing at least 1 trade (as a way of asserting the behavior)

Reply:
Okay, I've added a basic regression algorithm.  It really doesn't do much other than have in the statistics the number of expected trades be 4 - which matches the limit on the symbols.

I'm not sure how beneficial it is though, because it seems like the regression suite doesn't use actual data - and therefore, the historical look-up of open interest isn't providing anything of significance.

Do you have any suggestions?

Otherwise, is it worth keeping?

Reply:
>  It really doesn't do much other than have in the statistics the number of expected trades

That's okay 👍 my suggestion was adding it as a simple example and as a regression test (which asserts on the stats)

> doesn't use actual data

How/on what is it trading? Open-source data folder has GC future tick data available with open interest https://github.com/QuantConnect/Lean/tree/master/Data/future/comex/tick/gc, using the GC future in those available days should work for using the `OpenInterestFutureUniverseSelectionModel` and placing some trade?

Reply:
Okay, I've updated it again with an additional check.  Hopefully it's a bit more inline with the expected approach now.

It's trading gold, but it's test is to look at the contract selection.

Reply:
Looks great ty 👍